clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1085,12 +1085,19 @@ static target getAccessTarget(QualType FieldTy,
       AccTy->getTemplateArgs()[3].getAsIntegral().getExtValue());
 }
 
-// FIXME: Free functions must have void return type, be declared at file scope,
-// outside any namespaces, and with the SYCL_DEVICE attribute. If the
-// SYCL_DEVICE attribute is not specified this function is not entered since the
-// possibility of the function being a free function is ruled out already.
+static bool hasDependentExpr(Expr **Exprs, const size_t ExprsSize) {
```

Comment:
This is simple enough I'd probably make it a lambda along these lines:
```
auto hasDependentExpr = [](auto Args) {
  return llvm::any_of(Args, [](const Expr *E) {
    return E->isValueDependent() || E->isTypeDependent();
  });
};
```
(you can call it with `IRAttr->args()` to use the range-based APIs, which is cleaner)

Reply:
OK, I've made that change.